### PR TITLE
refactor(PLATFORM-4107): notify user to verify email after update

### PR DESCRIPTION
**Pull request by mc-jones**
This PR resolves [PLATFORM-4107]

### Description

This PR adds an alert message after a user updates their email instructing them to verify their email address. It supports the backend changes being made as part of this [ticket](https://artsyproduct.atlassian.net/browse/PLATFORM-4178).

#### iOS:
NOTE: I saw double notifications in other areas, so assuming this is unrelated to my changes. 
Screen Shot 2022-05-25 at 4 18 35 PM

#### Andorid:
Screen Shot 2022-05-26 at 1 15 36 PM

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Dev changes

- Alert user they need to confirm their email address after updating - Matt

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PLATFORM-4107]: https://artsyproduct.atlassian.net/browse/PLATFORM-4107?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mc-jones** [comment]
I could actually use a quick pair on adding a test for this. Thought it would be pretty straightforward, but ran into some relay issues when mocking out `useToast`.

File: src/app/Scenes/MyAccount/MyAccountEditEmail.tsx [hunk 1]
```diff
@@ -33,6 +36,13 @@ const MyAccountEditEmail: React.FC<{ me: MyAccountEditEmail_me; relay: RelayProp
       onSave={async (dismiss) => {
         try {
           await updateMyUserProfile({ email })
+
+          if (email !== me.email) {
+            toast.show("Please confirm your new email for this update to take effect", "bottom", {
```
Comment (brainbicycle) [inline src/app/Scenes/MyAccount/MyAccountEditEmail.tsx:41]: the double toast is a known issue that will be fixed with the new app shell, might suggest using a centered toast for the time being
```suggestion
            toast.show("Please confirm your new email for this update to take effect", "center", {
```
```suggestion
            toast.show("Please confirm your new email for this update to take effect", "bottom", {
```

Comment (brainbicycle) [inline src/app/Scenes/MyAccount/MyAccountEditEmail.tsx:41]: the double toast is a known issue that will be fixed with the new app shell, might suggest using a centered toast for the time being
```suggestion
            toast.show("Please confirm your new email for this update to take effect", "center", {
```
```suggestion
            toast.show("Please confirm your new email for this update to take effect", "bottom", {
```

**Review by brainbicycle — APPROVED** [review]
Looks good! 
The other thing to mention is we have an email confirmation banner that shows on Home Screen for unconfirmed accounts, I assume it will show again for this case but would be good to check once this in beta.

https://user-images.githubusercontent.com/49686530/170718193-345f1254-e518-48a1-8946-2d4ec2e00316.mov





code is here: https://github.com/artsy/eigen/blob/main/src/app/Scenes/Home/Components/EmailConfirmationBanner.tsx

File: src/app/Scenes/MyAccount/MyAccountEditEmail.tsx [hunk 2]
```diff
@@ -33,6 +36,13 @@ const MyAccountEditEmail: React.FC<{ me: MyAccountEditEmail_me; relay: RelayProp
       onSave={async (dismiss) => {
         try {
           await updateMyUserProfile({ email })
+
+          if (email !== me.email) {
+            toast.show("Please confirm your new email for this update to take effect", "middle", {
```
Comment (brainbicycle) [inline src/app/Scenes/MyAccount/MyAccountEditEmail.tsx:41]: whoops 😅 sorry

Comment (mc-jones) [reply src/app/Scenes/MyAccount/MyAccountEditEmail.tsx:41]: 😆  No worries!

**Comment by mc-jones** [comment]
>The other thing to mention is we have an email confirmation banner that shows on Home Screen for unconfirmed accounts, I assume it will show again for this case but would be good to check once this in beta.

Noted. We can double-check once in beta. Are you OK if we merge this and I follow up with the test separately?

**Comment by mc-jones** [comment]
Follow-up tests from @MounirDhahri in #6831.

**Pull request merged**